### PR TITLE
Adding uri_sans to pki_secret_backend_cert

### LINKED ISSUE #758: Support for uri_sans in vault_pki_secret_backend_cert
Related to https://github.com/terraform-providers/terraform-provider-vault/issues/372 but for certificates.

### Terraform Version
```
$ terraform -v
Terraform v0.12.24
+ provider.vault v2.10.0
```

### Affected Resource(s)
Please list the resources as a list, for example:
- `vault_pki_secret_backend_cert`

### Terraform Configuration Files
```hcl
resource "vault_pki_secret_backend_cert" "istio-ca" {
  backend       = "${var.root_backend_path}"
  name            = vault_pki_secret_backend_role.istio-ca.name

  common_name   = "Istio Intermediate CA"
  alt_names     = ["localhost"]
  uri_sans    = [
    "spiffe://cluster.local/ns/istio-system/sa/citadel",
    "spiffe://cluster.global/ns/istio-system/sa/citadel",
    "spiffe://cluster_name/ns/istio-system/sa/citadel"
  ]
}
```

### Expected Behavior
[Generate Certificate](https://www.vaultproject.io/api-docs/secret/pki#generate-certificate) should be called with uri_sans set.

### Actual Behavior
```
Error: vault_pki_secret_backend_cert.istio-ca: : invalid or unknown key: uri_sans
```

### Steps to Reproduce
Configure `vault_pki_secret_backend_cert` with a uri_sans (which isn't currently supported) and then terraform apply.

### References
- https://github.com/terraform-providers/terraform-provider-vault/issues/372

### PR DESCRIPTION
**Pull request by theorlandog**
Fixes [#758 ](https://github.com/terraform-providers/terraform-provider-vault/issues/758)

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #758 

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-vault/blob/master/CHANGELOG.md): 

```release-note
Add support for uri_sans parameter for resource vault_pki_secret_backend_cert
```

### CODE REVIEW

**Comment by bendrucker** [comment]
@theorlandog This PR has changed all the file permissions to 755. The previous value of 644 was correct. If you revert that (`chmod 644 vault/*.go` should work) you'll get the diff you expect here.

**Comment by theorlandog** [comment]
@bendrucker Sorry about that. Windows being weird. Just updated for you.

**Comment by bendrucker** [comment]
No worries, I was just here checking in on another PR and figured I'd mention it so you don't waste a review cycle with an actual maintainer. Which I'm not FWIW, wish I could help you get this merged. Looks like the line endings got converted to `\r\n` which still presents an extra diff.

Tweaking your git settings should help with that, specifically `autocrlf`:

https://help.github.com/en/github/using-git/configuring-git-to-handle-line-endings

**Comment by theorlandog** [comment]
@bendrucker ahh good looks. Definitely gotta spend the time to learn my git tricks some day. Thanks for helping me out.

File: vault/resource_pki_secret_backend_cert.go [hunk 1]
```diff
@@ -56,6 +56,15 @@ func pkiSecretBackendCertResource() *schema.Resource {
 					Type: schema.TypeString,
 				},
 			},
+			"uri_sans": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "List of alternative URIs.",
+				ForceNew:    true,
```
Comment (tyrannosaurus-becks) [inline vault/resource_pki_secret_backend_cert.go:63]: Should this really `ForceNew` if this field changes?

File: vault/resource_pki_secret_backend_cert.go [hunk 2]
```diff
@@ -167,6 +176,12 @@ func pkiSecretBackendCertCreate(d *schema.ResourceData, meta interface{}) error
 		ipSans = append(ipSans, iIpSan.(string))
 	}
 
+	iURISans := d.Get("uri_sans").([]interface{})
```
Comment (tyrannosaurus-becks) [inline vault/resource_pki_secret_backend_cert.go:179]: Could we simplify this code if we did something more like:
```
if raw, ok := d.GetOk("uri_sans"); ok {
    data["uri_sans"] = raw
}
```
Or is there a downside to that?

File: vault/resource_pki_secret_backend_cert.go (hunk 1 above)
Comment (theorlandog) [reply vault/resource_pki_secret_backend_cert.go:63]: Hi @tyrannosaurus-becks,
    To be quite honest, I lifted most of this code for a similar PR raised against CAs. https://github.com/terraform-providers/terraform-provider-vault/pull/373/files#diff-318e4d5fc8ea223f57928c837b6ea671R60. In that PR, it was set to ForceNew as well. I would think force new matches this matches the intent, as you probably need the uri set in the cert if your are taking the time to set it in the tf.

File: vault/resource_pki_secret_backend_cert.go (hunk 2 above)
Comment (theorlandog) [reply vault/resource_pki_secret_backend_cert.go:179]: Hi @tyrannosaurus-becks,
    Same idea as the last comment. I jacked most of this code from another PR. https://github.com/terraform-providers/terraform-provider-vault/pull/373/files#diff-318e4d5fc8ea223f57928c837b6ea671R193. It smells like your block would be equivalent, but maybe it makes sense to keep the two related blocks to use the same form? Not sure your opinion there.

**Review by tomhjp — APPROVED** [review]
LGTM, thanks for the contribution!

**Comment by catsby** [comment]
Hello all - this was released in [`v2.13.0`][v2130] , thanks for the contribution and your patience!


[v2130]: https://github.com/terraform-providers/terraform-provider-vault/releases/tag/v2.13.0

**Pull request merged**